libraries/resource_systemd_mount.rb

```diff
@@ -0,0 +1,21 @@
+require_relative 'resource_systemd_unit'
+require_relative 'systemd_mount'
+
+class Chef::Resource
+  class SystemdMount < Chef::Resource::SystemdUnit
+    self.resource_name = :systemd_mount
+    provides :systemd_mount
+
+    def unit_type(_ = nil)
+      :mount
+    end
+
+    Systemd::Mount::OPTIONS.each do |option|
+      attribute option.underscore.to_sym, kind_of: String, default: nil
```

Comment:
I think this is reasonable for the first version.  It might be nice in the future to allow ruby booleans in place of systemd's "yes"/"no" booleans.